client/landing/gutenboarding/components/mshots-image/index.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import classnames from 'classnames';
+import { Spinner } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+interface Props {
+	src: string;
+	alt: string;
+}
+
+const MshotsImage: React.FunctionComponent< Props > = ( { src, alt } ) => {
+	const [ resolvedUrl, setResolvedUrl ] = React.useState< string | undefined >();
+	const [ count, setCount ] = React.useState( 0 );
+	const [ isVisible, setIsVisible ] = React.useState( false );
+
+	const mShotsEndpointUrl = src;
+
+	React.useEffect( () => {
+		async function fetchMshots() {
+			const response = await window.fetch( mShotsEndpointUrl, {
+				method: 'GET',
+				mode: 'cors',
+				cache: 'no-cache',
+			} );
+
+			if ( response.ok && response.headers.get( 'Content-Type' ) === 'image/jpeg' ) {
+				setResolvedUrl( mShotsEndpointUrl );
+			}
+
+			if ( response.status === 307 ) {
+				const id = setTimeout( () => setCount( count + 1 ), 1000 );
+				return () => clearTimeout( id );
+			}
+		}
+		fetchMshots();
+	}, [ mShotsEndpointUrl, count ] );
+
+	return (
+		<div className="mshots-image__container">
+			{ ( ! resolvedUrl || ! isVisible ) && (
+				<p className="mshots-image__loader">
+					<Spinner />
```

Comment:
I know this might've been just temporary due lack of designs but let's use pulsating gray box instead of wall of spinners. 😁

Reply:
Yep! It was just a placeholder to use in place of real designs, and something that looked better than the string “loading...” 😄

Reply:
Updated to be pulsating gray box instead of spinners.